File: interviewbit-binary-search-square-root-of-integer.java
public class Solution {
    public int sqrt(int a) {
        int start = 0;
        int end = a;
        int mid;
        long squared;
        while (start <= end) {
            mid = (start + end) / 2;
            squared = mid * mid;
            if (squared == a) {
                return mid;
            } else if (squared < a) {
                start = mid + 1;
            } else {
                end = mid - 1;
            }
        }
        return start - 1;
    }
}

File: interviewbit-binary-search-square-root-of-integer.py
class Solution:
    # @param A : integer
    # @return an integer
    def sqrt(self, A):
        start = 0
        end = A
        while start <= end:
            mid = (start + end) / 2
            squared = mid ** 2
            if squared == A:
                return mid
            elif squared < A:
                start = mid + 1
            elif squared > A:
                end = mid - 1

        return start - 1
